### PR TITLE
Long-term fix for PRs from a fork, CODEOWNER file

### PR DESCRIPTION
**Pull request by jschuurk-kr**
relates to https://github.com/kiesraad/abacus/issues/513

Scope:
- fix retrieving contributors file from `"base"` instead of `"head"` (relevant in case of PRs from a fork)
- add a CODEOWNER file
- add reminder to README to run `npm build`

Idea behind the CODEOWNER file is that we give the Abacus dev team access to this repo, but require an approval from at least one of the people in the CODEOWNER file.

---

I tested the fix by adding this branch of cla-bot to a repo with both an internal and external PR: https://github.com/jschuurk-kr/gh-actions-playground/pulls Then I tried different permutations of the existence and the contents of `contributors.yml` in the `main` branch of this repo and the two PR branches.

### CODE REVIEW

**Comment by Lionqueen94** [comment]
The console.logs can be removed now before merging

**Comment by jschuurk-kr** [comment]
> The console.logs can be removed now before merging

I'd prefer to keep them. Otherwise if something weird happens, the only information we have is if the latest run of the cla-bot did or did not post a comment with certain usernames on the PR.

**Review by Lionqueen94 — APPROVED** [review]

**Review by praseodym — DISMISSED** [review]

**Comment by praseodym** [comment]
> Idea behind the CODEOWNER file is that we give the Abacus dev team access to this repo, but require an approval from at least one of the people in the CODEOWNER file.

Getting two approvals is going to be tricky now, though 🙃

**Comment by jschuurk-kr** [comment]
> > Idea behind the CODEOWNER file is that we give the Abacus dev team access to this repo, but require an approval from at least one of the people in the CODEOWNER file.
> 
> Getting two approvals is going to be tricky now, though 🙃

Ah, so the settings _"The number of approving reviews that are required before a pull request can be merged."_ and _"Require an approving review in pull requests that modify files that have a designated code owner."_ interact differently than I had thought/hoped.

**Comment by Lionqueen94** [comment]
> > > Idea behind the CODEOWNER file is that we give the Abacus dev team access to this repo, but require an approval from at least one of the people in the CODEOWNER file.
> > 
> > 
> > Getting two approvals is going to be tricky now, though 🙃
> 
> Ah, so the settings _"The number of approving reviews that are required before a pull request can be merged."_ and _"Require an approving review in pull requests that modify files that have a designated code owner."_ interact differently than I had thought/hoped.

So all Kiesraad devs should be CODEOWNERS then? :sweat_smile:

**Comment by jschuurk-kr** [comment]
@praseodym @ring-ring-ring Kunnen jullie deze PR appproven, zodat die gemerged kan worden?

**Comment by Lionqueen94** [comment]
@jschuurk-kr It seems like there's an issue with adding Niels to the CODEOWNERS file:
image

**Comment by jschuurk-kr** [comment]
> @jschuurk-kr It seems like there's an issue with adding Niels to the CODEOWNERS file: image

I added Niels to the Abacus developers team in GitHub after making that commit. And I don't see a way to make GitHub re-check this... So should be fine. :crossed_fingers:

**Comment by jschuurk-kr** [comment]
@ring-ring-ring @praseodym Can you please approve this cla-bot PR?

**Comment by praseodym** [comment]
I just gave @ring-ring-ring explicit write permissions on this repository to fix the `CODEOWNERS` file error:
Screenshot 2024-11-20 at 19 01 08
Screenshot 2024-11-20 at 19 02 50

**Review by praseodym — APPROVED** [review]

**Review by ring-ring-ring — APPROVED** [review]

**Pull request merged**
